great_expectations/cli/datasource.py

```diff
@@ -134,17 +147,23 @@ def _add_sqlalchemy_datasource(context):
 
     credentials = {}
     while True:
-        cli_message(msg_sqlalchemy_config_connection.format(data_source_name))
+        cli_message(msg_db_config.format(data_source_name))
 
         if selected_database == SupportedDatabases.MYSQL:
+            if not load_library("pymysql"):
+                return None
             # TODO implement
             pass
         elif selected_database == SupportedDatabases.POSTGRES:
             credentials = _collect_postgres_credentials(default_credentials=credentials)
         elif selected_database == SupportedDatabases.REDSHIFT:
+            if not load_library("psycopg2"):
```

Comment:
The mapping between supported databases and libraries here is slightly confusing; psycopg2 isn't the usual redshift driver (but it is required by it).